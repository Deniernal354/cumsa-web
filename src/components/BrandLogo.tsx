"use client";
import Image from "next/image";
import { useState } from "react";

type Props = { size?: number; className?: string };

export default function BrandLogo({ size = 44, className = "" }: Props) {
  const [src, setSrc] = useState("/CU-MSA-LOGO-blank-2.png");
  const dim = { width: size, height: size } as const;
  const isBlank = src.includes("CU-MSA-LOGO-blank-2.png");
  if (isBlank) {
    return <span aria-hidden className={`logo-mask-blue rounded-sm ${className}`} style={dim} />;
  }
  return (
    <Image
      src={src}
      alt="CUMSA logo"
      width={size}
      height={size}
      priority
      sizes={`${size}px`}
      onError={() => setSrc("/CU-MSA-LOGO-blank-2.png")}
      className={`object-contain rounded-sm ${className}`}
      style={dim}
    />
  );
}
