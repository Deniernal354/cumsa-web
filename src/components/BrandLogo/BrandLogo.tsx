// ...existing code from BrandLogo.tsx...
"use client";
import Image from "next/image";
import { useState } from "react";
import { LOGOS } from "@/lib/constants";

type Props = { size?: number; className?: string };

export default function BrandLogo({ size = 44, className = "" }: Props) {
  const [src, setSrc] = useState<string>(LOGOS.main);
  const dim = { width: size, height: size } as const;

  return (
    <Image
      src={src}
      alt="CUMSA logo"
      width={size}
      height={size}
      priority
      sizes={`${size}px`}
      onError={() => setSrc(LOGOS.fallback)}
      className={`object-contain rounded-sm ${className}`}
      style={dim}
    />
  );
}
