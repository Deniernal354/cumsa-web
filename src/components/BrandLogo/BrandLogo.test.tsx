import { render } from "@testing-library/react";
import BrandLogo from "./BrandLogo";

test("BrandLogo snapshot", () => {
  const { asFragment } = render(<BrandLogo />);
  expect(asFragment()).toMatchSnapshot();
});
