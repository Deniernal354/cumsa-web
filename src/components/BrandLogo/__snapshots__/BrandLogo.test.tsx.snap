// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo snapshot 1`] = `
<DocumentFragment>
  <span
    aria-hidden="true"
    class="logo-mask-blue rounded-sm "
    style="width: 44px; height: 44px;"
  />
</DocumentFragment>
`;
